tests/rptest/clients/kcl.py

```diff
@@ -182,6 +187,89 @@ def describe_topic(self,
 
         return self._cmd(cmd, attempts=1)
 
+    def alter_partition_reasssignments(self, topics: dict[str,
+                                                          dict[int,
+                                                               list[int]]]):
+        """
+        :param topics: the key is a topic and the value is a dict that maps partition IDs
+                       to new replica assignments
+        :return: list of KclAlterPartitionReassignmentsResponse
+        """
+        cmd = ["admin", "partas", "alter"]
+
+        for topic in topics:
+            assert len(topics[topic]) > 0
+            reassignment_str = f"{topic}:"
+            partitions = []
+            for pid in topics[topic]:
+                part_str = f"{pid}"
+                if len(topics[topic][pid]) > 0:
+                    part_str = f"{pid}->{topics[topic][pid]}"
+                    # Remove empty and [] characters
+                    part_str = part_str.replace('[', '').replace(']', '')
+                    part_str = part_str.replace(' ', '')
+                partitions.append(part_str)
+            join_partitions = ";".join(partitions)
+            reassignment_str += join_partitions
+            cmd.append(reassignment_str)
+
+        lines = self._cmd(cmd, attempts=1).splitlines()
+        res_re = re.compile(
+            r"^(?P<topic>[a-z\-]+?) +(?P<partition>[0-9]+?) +(?P<status>[a-zA-Z]+?)$"
+        )
+        for l in lines:
+            l = l.strip()
+            m = res_re.match(l)
+            assert m is not None
+
+        return lines
+
+    def list_partition_reasssignments(self,
+                                      topics: Optional[dict[
+                                          str, list[int]]] = None):
+        """
+        :param topics: dict where topic name is the key and the value is the list
+                       of partition IDs
+        :return: list of KclListPartitionReassignmentsResponse
+        """
+        cmd = ["admin", "partas", "list"]
+        lines = None
+        if topics is None:
+            lines = self._cmd(cmd, attempts=1).splitlines()
+        else:
+            for topic in topics:
+                topic_str = f"{topic}:{topics[topic]}"
+                # Remove empty and [] characters
+                topic_str = topic_str.replace('[', '').replace(']', '')
+                topic_str = topic_str.replace(' ', '')
+                cmd.append(topic_str)
+
+            lines = self._cmd(cmd, attempts=1).splitlines()
```

Comment:
same here, just return the value

Reply:
> same here, just return the value

See the return values for list_offsets and offset_for_leader_epoch in the KCL class. They also parse the output and return objects.